### DISCUSSION TITLE
Sveltekit demo - why is `locals.LL` not used in layout.ts

### DISCUSSION THREAD

**Discussion by gterras** [ivanhofer/typesafe-i18n#508] [Q&A] [answered]
Hi, 

Is there a specific reason for `locals.LL` not to be used in https://github.com/ivanhofer/typesafe-i18n-demo-sveltekit/blob/main/src/routes/%2Blayout.ts ? 

I don't really understand the hassle of importing LL in this file since `+layout.server.ts` could have returned `locals.LL`. 

Is this for performance reasons? Or reactivity reason in case of a locale change? 

Thanks

**Comment by gterras** [comment, marked as answer]
`Cannot stringify a function (data.LL)` might be the reason why

**Reply by ivanhofer** [reply]
You can only access `locals` in `.server.ts` files. 
The reason to not returning all translations via `data` props is simple: SvelteKit would need to serialize your full dictionary and your browser would download it again each time you refresh the page. Doing it via the `loadLocaleAsync` function, makes sure that a real `NetworkRequest` get's made. The dictionary file will be cached by your browser until you modify it and the hash of that file changes (handled automatically by `vite`).

From a performance standpoint the SvelteKit implementation is highly optimized. Other platforms like NextJS don't offer an elegant solution to the problem of loading dictionaries while offering SSR.

**Reply by gterras** [reply]
Thanks for this this makes sense. After reading a few threads about this (including the i18n thread on sveltekit repo and what you said there) and realizing SSRed translation is a complicated topic I think I will stick to in-component translations since it is easy to accidentally mess it up in load function (forgetting parent await or such). The dot notation access you wrote in my other thread will greatly help for this.
